src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,429 @@
+                  return {
+                    value: domainType,
+                    label: domainType.category,
+                  };
+                }, datasetDetails.domainTypes),
+              },
+              {
+                label: 'Program Data',
+                options: _.map((programDataType) => {
+                  return {
+                    value: programDataType,
+                    label: programDataType.name,
+                  };
+                }, datasetDetails.programDataTypes),
+              },
+            ],
+            'aria-label': 'add criteria',
+            placeholder: 'Add criteria',
+            value: undefined,
+            onChange: (x) => {
+              // FIXME: is there a way to remove any?
+              const criteria = createCriteriaFromType((x as any).value);
+              updateCohort(_.set(`criteriaGroups.${index}.criteria.${criteriaGroup.criteria.length}`, criteria));
+            },
+          }),
+        ]),
+      ]),
+      div(
+        {
+          style: {
+            paddingTop: 5,
+            marginTop: 10,
+            marginBottom: 10,
+            borderTop: `1px solid ${colors.dark(0.35)}`,
+            display: 'flex',
+            justifyContent: 'flex-end',
+            fontWeight: 'bold',
+          },
+        },
+        [div({ style: { marginRight: 10 } }, [`Group count: ${criteriaGroup.count}`])]
+      ),
+    ]
+  );
+};
+
+const RenderCohort = ({
+  datasetDetails,
+  cohort,
+  updateCohort,
+}: {
+  cohort: Cohort | undefined;
+  datasetDetails: DatasetResponse;
+  updateCohort: CohortUpdater;
+}) => {
+  return div({ style: { width: '47rem' } }, [
+    cohort == null
+      ? 'No cohort found'
+      : div([
+          _.map(
+            ([index, criteriaGroup]) =>
+              h(Fragment, { key: criteriaGroup.name }, [
+                h(CriteriaGroupView, { index, criteriaGroup, updateCohort, cohort, datasetDetails }),
+                div({ style: { marginTop: '1rem', display: 'flex', alignItems: 'center' } }, [
+                  div(
+                    {
+                      style: {
+                        height: 45,
+                        width: 45,
+                        backgroundColor: 'white',
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                        borderRadius: '50%',
+                        border: `1px solid ${colors.dark(0.35)}`,
+                        fontStyle: 'italic',
+                      },
+                    },
+                    ['And']
+                  ),
+                  div({
+                    style: { marginLeft: 5, flexGrow: 1, borderTop: `1px solid ${colors.dark(0.35)}` },
+                  }),
+                ]),
+              ]),
+            Utils.toIndexPairs(cohort.criteriaGroups)
+          ),
+        ]),
+  ]);
+};
+
+const editorBackgroundColor = colors.light(0.7);
```

Comment:
Do different pages have different background colors? If not, we might want to think about putting this one level higher in the `DatasetBuilder.ts` container.

Reply:
I moved this background color setting to datsetbuilder so all subpages should now share the same background color.